### PR TITLE
nushell: add module

### PR DESCRIPTION
**Pull request by Philipp-M**
### Description

Add module for https://github.com/nushell/nushell

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/rycee/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/nushell.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      type = types.package;
+      default = pkgs.nushell;
+      defaultText = literalExample "pkgs.nushell";
+      description = "The package to use for nushell.";
+    };
+
+    settings = mkOption {
+      type = with types;
+        let
+          prim = either bool (either int str);
```
Comment (berbiche) [inline modules/programs/nushell.nix:36]: Isn't this the same as `oneOf [ bool int str ]` ?

Comment (Philipp-M) [reply modules/programs/nushell.nix]: Thanks, I updated this with your suggestion.
To be honest, I copied a lot from a similar derivation (*starship*, which is used automatically by nushell btw.)

**Comment by Philipp-M** [comment]
nushell 0.16 is in nixos-unstable (already in nixpkgs-channels) so it should be ready for merge.

**Comment by teto** [comment]
tried locally and overall this looks ok but I got this issue
```
Welcome to Nushell 0.16.0 (type 'help' for more info)error: ls unexpected -l  ┌─ shell:1:19
  │
1 │ alias ll [] { ls -l }
  │                   ^ unexpected flag (try ls -h)

```
I suspect it's due to one of my alias using an ls option unsupported by nushell ?

**Comment by teto** [comment]
just tested and seems like it. Also even though I am not fond of the feature, we usually add new modules in news.nix, could you add an entry for your module please ?

**Comment by colemickens** [comment]
This is working for me. Thanks for adding this!

**Comment by Philipp-M** [comment]
> I suspect it's due to one of my alias using an ls option unsupported by nushell ?

Oops, I just assumed, that `ls -l` exists in nu. Corrected this with `alias la [] { ls -a }` instead.
And added an entry to `news.nix`

**Pull request merged**